src/Microsoft.AspNet.Mvc.TagHelpers/EnvironmentTagHelper.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting &lt;environment&gt; elements that conditionally renders content
+    /// based on the current value of <see cref="IHostingEnvironment.EnvironmentName"/>.
+    /// </summary>
+    public class EnvironmentTagHelper : TagHelper
+    {
+        private static readonly char[] _nameSeparator = new[] { ',' };
+
+        /// <summary>
+        /// A comma separated list of environment names in which the content should be rendered.
+        /// </summary>
+        public string Names { get; set; }
```

Comment:
Should add a remark stating that names are compared case insensitively to the environment name.